### DISCUSSION TITLE
prNotPendingHours not honoured

### DISCUSSION THREAD

**Discussion by martinbonnin** [renovatebot/renovate#9118] [Request Help] [answered]
**What Renovate type, platform and version are you using?**

Using the hosted renovate App on https://github.com/apollographql/apollo-android/ (renovate.json [here](https://github.com/apollographql/apollo-android/blob/main/renovate.json))

**Describe the bug**

prNotPendingHours is not honoured. Despite having set `prNotPendingHours=1`, the PR never gets created. The logs show:

```
Branch is 3 hours old - skipping PR creation
```

**Relevant debug logs**

From https://app.renovatebot.com/dashboard#github/apollographql/apollo-android/316803136:

```
Checking branch combined status(branch="renovate/documentation-infrastructure")
DEBUG: getBranchStatus(renovate/documentation-infrastructure)(branch="renovate/documentation-infrastructure")
DEBUG: branch status check result(branch="renovate/documentation-infrastructure")
{
  "state": "pending",
  "statuses": []
}
DEBUG: No check runs found(branch="renovate/documentation-infrastructure")
{
  "result": {
    "total_count": 0,
    "check_runs": []
  }
}
DEBUG: getBranchStatus() result(branch="renovate/documentation-infrastructure")
{
  "branchStatus": "yellow",
  "branchName": "renovate/documentation-infrastructure"
}
DEBUG: Branch status is "yellow" - checking timeout(branch="renovate/documentation-infrastructure")
DEBUG: Branch is 3 hours old - skipping PR creation(branch="renovate/documentation-infrastructure")
DEBUG: processBranch with 1 upgrades(branch="renovate/documentation-theme")
DEBUG: Setting current branch to main

```

Dashboard: https://github.com/apollographql/apollo-android/issues/2835

Clicking the box schedules the PR correctly but the PR never gets created automatically.

**Comment by rarkins** [comment]
According to this log, you don't have any check runs or status checks being run on the branch. Can you confirm if that's correct?

So is your goal that you set prNotPendingHours=1 so that the branch times out with no tests (i.e. pending) and then creates a PR so that tests can be run?

**Reply by martinbonnin** [reply]
> According to this log, you don't have any check runs or status checks being run on the branch. Can you confirm if that's correct?

I'm not 100% sure how to check this on branches (vs pull requests where they show in the pull request) but I believe that's correct. We have a github action workflow that we disable for updates to docs (that typically come from renovate) to save some Action time. 

> So is your goal that you set prNotPendingHours=1 so that the branch times out with no tests (i.e. pending) and then creates a PR so that tests can be run?

Yep exactly.

**Comment by rarkins** [comment, marked as answer]
Tracing the logic, it passes this line: https://github.com/renovatebot/renovate/blob/926f23f786aac51242430152d2f9759ef90c6d5b/lib/workers/pr/index.ts#L209

And then: https://github.com/renovatebot/renovate/blob/926f23f786aac51242430152d2f9759ef90c6d5b/lib/workers/pr/index.ts#L223

So either `elapsedHours < config.prNotPendingHours` (in which case`config.prNotPendingHours` is >= 3 and not 1 as we think) or `config.stabilityStatus !== BranchStatus.yellow`.

I'm not sure why that is but have added a non-null check to config.stabilityStatus.

**Reply by martinbonnin** [reply]
Looks like that would work, thanks! Our renovate jobs are still using `24.81.0` and the fix is in `24.81.1`, can I force the renovate version to `24.81.1` somewhere or is it going to be updated automatically at some point?

**Reply by rarkins** [reply]
It will be updated automatically soon

**Reply by martinbonnin** [reply]
That worked 🎉: https://github.com/apollographql/apollo-android/pull/3001
Thanks!
